File: poc/InkPoc/InkPoc/Services/Ink/InkZoomService.cs
using System;
using Windows.Foundation;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;

namespace InkPoc.Services.Ink
{
    public class InkZoomService
    {
        private const float defaultZoomFactor = 0.1f;
        private readonly ScrollViewer scrollViewer;

        public InkZoomService(ScrollViewer _scrollViewer)
        {
            scrollViewer = _scrollViewer;
        }

        public float ZoomIn(float zoomFactor = defaultZoomFactor) => ExecuteZoom(scrollViewer.ZoomFactor + zoomFactor);

        public float ZoomOut(float zoomFactor = defaultZoomFactor) => ExecuteZoom(scrollViewer.ZoomFactor - zoomFactor);

        public float ResetZoom() => ExecuteZoom(1f);

        public void FitToScreen()
        {
            if (scrollViewer.Content is FrameworkElement element)
            {
                FitToSize(element.Width, element.Height);
            }
        }

        public void FitToSize(double width, double height)
        {
            if (width == 0 || height == 0)
            {
                return;
            }

            var ratioWidth = scrollViewer.ViewportWidth / width;
            var ratioHeight = scrollViewer.ViewportHeight / height;
            var zoomFactor = (float)(Math.Min(ratioWidth, ratioHeight));

            ExecuteZoom(zoomFactor);
        }

        private float ExecuteZoom(float zoomFactor)
        {
            if(scrollViewer.ChangeView(scrollViewer.HorizontalOffset, scrollViewer.VerticalOffset, zoomFactor))
            {
                return zoomFactor;
            }

            return scrollViewer.ZoomFactor;
        }
    }
}

File: templates/Uwp/_composition/_shared/Page.Ink.AddZoomServiceVB/Services/Ink/InkZoomService.vb
Namespace Services.Ink
    Public Class InkZoomService
        Private Const DefaultZoomFactor As Single = 0.1F
        Private ReadOnly _scrollViewer As ScrollViewer

        Public Sub New(scrollViewer As ScrollViewer)
            _scrollViewer = scrollViewer
        End Sub

        Public Function ZoomIn(Optional zoomFactor As Single = DefaultZoomFactor) As Single
            Return ExecuteZoom(_scrollViewer.ZoomFactor + zoomFactor)
        End Function

        Public Function ZoomOut(Optional zoomFactor As Single = DefaultZoomFactor) As Single
            Return ExecuteZoom(_scrollViewer.ZoomFactor - zoomFactor)
        End Function

        Public Function ResetZoom() As Single
            Return ExecuteZoom(1F)
        End Function

        Public Sub FitToScreen()
            Dim element As FrameworkElement = TryCast(_scrollViewer.Content, FrameworkElement)

            If element IsNot Nothing Then
                FitToSize(element.Width, element.Height)
            End If
        End Sub

        Public Sub FitToSize(width As Double, height As Double)
            If width = 0 OrElse height = 0 Then
                Return
            End If

            Dim ratioWidth = _scrollViewer.ViewportWidth / width
            Dim ratioHeight = _scrollViewer.ViewportHeight / height
            Dim zoomFactor = CSng(Math.Min(ratioWidth, ratioHeight))
            ExecuteZoom(zoomFactor)
        End Sub

        Private Function ExecuteZoom(zoomFactor As Single) As Single
            If _scrollViewer.ChangeView(_scrollViewer.HorizontalOffset, _scrollViewer.VerticalOffset, zoomFactor) Then
                Return zoomFactor
            End If

            Return _scrollViewer.ZoomFactor
        End Function
    End Class
End Namespace
